lib/HLSL/HLOperationLower.cpp

```diff
@@ -829,7 +829,11 @@ Constant *GetLoadInputsForEvaluate(Value *V, std::vector<CallInst*> &loadList) {
     // TODO: We are assuming that the operand of insertelement is a LoadInput.
     // This will fail on the case where we pass in matrix member using array subscript.
     while (!isa<UndefValue>(Vec)) {
-      InsertElementInst *insertInst = cast<InsertElementInst>(Vec);
+      InsertElementInst *insertInst = dyn_cast<InsertElementInst>(Vec);
+      if (!insertInst) {
+        Vec->getContext().emitError(kAttributeEvaluateError);
+        break;
+      }
       Vec = insertInst->getOperand(0);
       Value *Elt = insertInst->getOperand(1);
       if (IsValidLoadInput(Elt)) {
```

Comment:
Can there be a case where only some of the elements are loadinputs? Something like this?

float main(float foo : IN) : OUT {
  float2 bar = float2(foo, 42);
}

Reply:
`EvaluateAttributeAtSample` must be given directly the attribute AFAIK.

Reply:
Could it be Shuffle which generated from swizzle like foo.zx when foo is float4?

Reply:
@adam-yang It's incorrect but I was wondering how the code would behave in that case.

Reply:
@python3kgae As it is now, we cannot do `EvaluateAttributeAtSample(foo.yx)`. Should this be allowed?

Reply:
We can do it later.